### PR TITLE
support iot workbench to configure devkit in remote development

### PR DESCRIPTION
**Pull request by RongXiefdu**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -18,6 +18,118 @@ interface Volume {
   name?: string | undefined;
 }
 
+interface PortOption {
+  baudRate: number | undefined;
```
Comment (Sneezry) [inline src/index.ts]: Use `baudRate?: number` to define optional member

File: src/index.ts [hunk 2]
```diff
@@ -18,6 +18,118 @@ interface Volume {
...
+}
+
+export class SerialPort {
+  static async getPlatform() {
+    return (await vscode.commands.executeCommand(
+      'iotcube.serialportGetPlatform'
+    )) as string;
+  }
+
+  static async chooseCOM(board: Board | undefined) {
```
Comment (Sneezry) [inline src/index.ts]: User `board?: Board` as optional parameter

File: src/index.ts [hunk 3]
```diff
@@ -19,31 +19,12 @@ interface Volume {
...
-  id: string;
-  model?: string;
-  detailInfo: string;
-  defaultBaudRate?: number;
-  vendorId?: string;
-  productId?: string;
-  exampleUrl?: string;
-  helpUrl?: string;
-  installation?: BoardInstallation;
+  baudRate?: number | undefined;
```
Comment (Sneezry) [inline src/index.ts]: If `baudRate` is optional, you can remove `undefined` from its type

File: src/index.ts [hunk 4]
```diff
@@ -53,18 +34,17 @@ export class SerialPort {
     )) as string;
   }
 
-  static async chooseCOM(board: Board | undefined) {
+  static async getComList() {
```
Comment (Sneezry) [inline src/index.ts]: Define type for `ports`

**Review by dilin-MS — APPROVED** [review]

File: src/index.ts [hunk 5]
```diff
@@ -18,6 +18,108 @@ interface Volume {
...
+  baudRate?: number;
+  dataBits?: number;
+  stopBits?: number;
+  xon?: boolean;
+  xoff?: boolean;
+  parity?: string;
+}
+
+export interface ComPort {
+  comName: string,
```
Comment (Sneezry) [inline src/index.ts]: use `;` here

**Review by Sneezry — APPROVED** [review]

File: src/index.ts (hunk 1 above)
Comment (RongXiefdu) [reply src/index.ts]: Fixed.

File: src/index.ts (hunk 3 above)
Comment (RongXiefdu) [reply src/index.ts]: Fixed.

File: src/index.ts (hunk 4 above)
Comment (RongXiefdu) [reply src/index.ts]: Fixed.

File: src/index.ts (hunk 5 above)
Comment (RongXiefdu) [reply src/index.ts]: Fixed.

**Pull request merged**